src/mlpack/methods/ann/layer/linear.hpp

```diff
@@ -31,7 +32,8 @@ namespace ann /** Artificial Neural Network. */ {
  */
 template <
     typename InputDataType = arma::mat,
-    typename OutputDataType = arma::mat
+    typename OutputDataType = arma::mat,
+    typename RegularizerType = L2Regularizer
```

Comment:
Do you think we should provide a class ``NoRegularizer`` and use that as the default instead of using ``RegularizerType(0)``, might avoid an unnecessary computation.